File: MovieProject/MovieProject.Web/Modules/Northwind/SalesInvoiceItems/SalesInvoiceItemsForm.cs
using Serenity.ComponentModel;
using System;

namespace MovieProject.Northwind.Forms;

[FormScript("Northwind.SalesInvoiceItems")]
[BasedOnRow(typeof(SalesInvoiceItemsRow), CheckNames = true)]
public class SalesInvoiceItemsForm
{
    public DateTime Date { get; set; }
    public string Notes { get; set; }
    public int SalesInvoiceId { get; set; }
}

File: MovieProject/MovieProject.Web/Modules/ServerTypes/Northwind/SalesInvoiceItemsForm.ts
import { DateEditor, StringEditor, ServiceLookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface SalesInvoiceItemsForm {
    Date: DateEditor;
    Notes: StringEditor;
    SalesInvoiceId: ServiceLookupEditor;
}

export class SalesInvoiceItemsForm extends PrefixedContext {
    static readonly formKey = 'Northwind.SalesInvoiceItems';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SalesInvoiceItemsForm.init)  {
            SalesInvoiceItemsForm.init = true;

            var w0 = DateEditor;
            var w1 = StringEditor;
            var w2 = ServiceLookupEditor;

            initFormType(SalesInvoiceItemsForm, [
                'Date', w0,
                'Notes', w1,
                'SalesInvoiceId', w2
            ]);
        }
    }
}